### ISSUE TITLE
"IsADirectoryError: [Errno 21] Is a directory" while importing module

### ISSUE THREAD

**Issue by peawyoyoyin** [evhub/coconut#861]
Minimal repository to reproduce the issue: https://github.com/peawyoyoyin/coconut-issue

How to reproduce (also in the readme file in the above repository):
1. Make sure python 3.12 and coconut 3.1.2 is installed
2. Clone the repository
3. With the repository root as the working directory, run `python3.12 -m coconut -r main.coco`

Error message:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 287, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/home/peawyoyoyin/coco-bug/main.py", line 2977, in <module>
    from utils import get_text  #1 (line in Coconut source)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 991, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1128, in get_code
  File "<frozen importlib._bootstrap_external>", line 1186, in get_data
IsADirectoryError: [Errno 21] Is a directory: '<REDACTED>/coco-bug/utils/__coconut_cache__'
```

Running the compiled output directly using `python3.12 main.py` also does not work, seem like python could not find the additional module.

My machine is running Ubuntu 22.04.5 LTS on WSL. I also tried similar setup on Windows 10 also got a similar error (but with "Permission Denied" error message instead).

I am willing to contribute on this fix if bandwidth is an issue, but please give me some pointers as it would be my first contribution to the repository 😄.

Lastly, I want to say I like the language so far, it is a great addition to python and in my scripting needs. Thanks for all the work, and looking forward to helping with the issue!

**Comment by evhub** [comment]
Thanks for the report! As a temporary fix, this should work if you set the environment variable `COCONUT_USE_CACHE_DIR = False`.

**Comment by evhub** [comment]
This is now fixed on `coconut-develop>=3.1.2-post_dev6` (`pip uninstall coconut && pip install -U coconut-develop>=3.1.2-post_dev6`).
